Apps/Src/Task_LogInfo.c

```diff
@@ -8,11 +8,10 @@ void Task_LogInfo(void *p_arg) {
     (void)p_arg;
 
     OS_ERR err;
-    EEPROM_Init();
 
     while (1){
         //sends percentage of charge left in the battery pack to EEPROM
-        EEPROM_LogData(DATA_SOC, Charge_GetPercent());
+        EEPROM_SetCharge(Charge_GetPercent());
```

Comment:
Is it necessary to have this in another task? Would it make more sense to send the charge data in the amps task but have a counter that counts down the number of times the Amps task runs before the EEprom is written to(to prevent too many writes)? Also should EEPROM_Init be run here?

Reply:
I feel like we would get issues with being interrupted in the middle of a write. According to datasheet fastest speed is around 400kHz.

Reply:
The advantage of putting this in its own task is that if the EEPROM function takes too long or gets stuck, it won't affect the amperes sampling rate. I am guessing that this function call will take at least 15ms because I think the EEPROM needs to wait 5ms in between byte writes, but we can profile it if we want.

Yeah, we might have issues when we are interrupted. I can add in error handling and see if any of the I2C messages are failing